src/basic/src/ScalarFunction.C

```diff
@@ -35,8 +38,27 @@ BaseScalarFunction<V, M>::BaseScalarFunction(const char * prefix,
     const VectorSet<V, M> & domainSet)
   : m_env(domainSet.env()),
     m_prefix((std::string)(prefix) + "func_"),
-    m_domainSet(domainSet)
+    m_domainSet(domainSet),
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+    m_parser(new BoostInputOptionsParser(m_env.optionsInputFileName())),
+#endif
+    m_fdStepSize(QUESO_BASESCALARFN_FD_STEPSIZE_ODV)
 {
+  // Snarf fd step size from input file.
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+  m_parser->registerOption<double>(m_prefix + "fdStepSize",
+                                   QUESO_BASESCALARFN_FD_STEPSIZE_ODV,
+                                   "step size for finite difference");
+  m_parser->scanInputFile();
+  m_parser->getOption<double>(m_prefix + "fdStepSize", m_fdStepSize);
+#else
+  m_fdStepSize = m_env.input()(m_prefix + "fdStepSize",
+                               QUESO_BASESCALARFN_FD_STEPSIZE_ODV);
+#endif
```

Comment:
Here's a compromise: make the finite difference step size a vector.  If the user sets a scalar in the input file then it'll set every component of the finite difference step size vector, but I won't bother adding logic to set individual components in the input file (it's a little more work).  Instead, I'll add some member functions that let you set individual component of the finite difference step size vector.